backend/portal/views.py

```diff
@@ -213,3 +215,45 @@ def get(self, request, id):
                 "poll_statistics": get_demographic_breakdown(id),
             }
         )
+
+
+class Posts(viewsets.ModelViewSet):
+    """
+    browse:
+    returns a list of Posts that are targeted at the current user.
+    Admins sees all the posts(?)
+
+    create:
+    Create a Post
+
+    partial_update:
+    Update certain fields in the Post.
+    Need to be the admin or the creator.
+
+    destroy:
+    Delete a Post.
+    Need to be the admin or the creator.
+    """
+
+    serializer_class = PostSerializer
+    permission_classes = [PostOwnerPermission | IsSuperUser]
+
+    def get_queryset(self):
+        return Post.objects.all()
+
+    @action(detail=False, methods=["get"])
+    def browse(self, request):
+        """
+        Returns a list of all posts that are targeted at the current user
+        Admins sees all the posts(?)
+        """
+        posts = (
+            Post.objects.all()
```

Comment:
we might need to change this query, as the get_user_populations function is not necessarily working atm